### PR TITLE
Add Maven extension support

### PR DESCRIPTION
**Pull request by autonomousapps**
I have verified this works with a local TC instance, and all the tests pass. It's WIP because I would like to add some test functionality around Maven scans. I'm opening this PR now in case you have any objection to the current implementation. The thing I currently like least is the small duplication with the regex in the `Util` class, but maybe it's acceptable.

### CODE REVIEW

**Comment by autonomousapps** [comment]
Also, I bumped the Gradle version because I had an issue publishing scans beforehand.

**Comment by etiennestuder** [comment]
Thanks, @autonomousapps . I will take a look and get back to you.

**Comment by autonomousapps** [comment]
Added tests around new functionality.

**Comment by autonomousapps** [comment]
@etiennestuder I addressed your comment about `equals` vs `contains` and also manually verified that it still works for capturing scans from the maven extension. (I did not manually test gradle build scans.)

**Comment by etiennestuder** [comment]
Thanks, @autonomousapps !

File: src/main/java/nu/studer/teamcity/buildscan/internal/Util.java [hunk 1]
```diff
@@ -6,7 +6,11 @@
 
 final class Util {
 
-    private static final Pattern BUILD_SCAN_URL_PATTERN = Pattern.compile("https?://.*/s/(.*)");
+    // Includes:
+    // 1. Optional `[INFO] ` prefix, which is present in Maven scans
+    // 2. A capturing group for the full URL (necessary for Maven scans)
+    // 3. A capturing group for the scan ID (necessary for the `BuildScanReference` instance).
+    private static final Pattern BUILD_SCAN_URL_PATTERN = Pattern.compile("(?:\\[INFO] )?(https?://.*/s/(.*))");
```
Comment (etiennestuder) [inline src/main/java/nu/studer/teamcity/buildscan/internal/Util.java]: @autonomousapps what is the `?:` for in this reg exp?

Comment (autonomousapps) [reply src/main/java/nu/studer/teamcity/buildscan/internal/Util.java]: `?:`, when placed at the start of a group, indicates a non-capturing group. I can index the other two groups as if the first wasn't there.

Comment (etiennestuder) [reply src/main/java/nu/studer/teamcity/buildscan/internal/Util.java]: Ah cool. Thanks for the quick and insightful response.

**Pull request merged**